### PR TITLE
feat: support wandb project env

### PR DESCRIPTION
**Pull request by firefighter-eric**
This PR allows the W&B logger to respect the WANDB_PROJECT
  environment variable.

  Previously, WandbLogger always used the explicit
  project_name argument passed to wandb.init(), which
  defaulted to DiffSynth-Studio. This meant that setting
  WANDB_PROJECT in the environment did not change the W&B
  project name.

  With this change, WandbLogger reads WANDB_PROJECT before
  calling wandb.init(), while preserving the existing default
  behavior when the environment variable is not set.

  Changes:

  - Read WANDB_PROJECT in diffsynth/diffusion/logger.py
  - Keep the default project name as DiffSynth-Studio
  - No changes to run name behavior or other logger behavior

  Validation:

  - Ran python3 -m py_compile diffsynth/diffusion/logger.py

  这个 PR 让 W&B logger 支持读取 WANDB_PROJECT 环境变量。

  之前 WandbLogger 会把显式传入的 project_name 传给
  wandb.init()，默认值是 DiffSynth-Studio。因此即使用户在环境
  变量里设置了 WANDB_PROJECT，W&B project 名称也不会被覆盖。

  这个改动会在调用 wandb.init() 前读取 WANDB_PROJECT。如果没
  有设置该环境变量，则保持原来的默认行为不变。

  变更内容：

  - 在 diffsynth/diffusion/logger.py 中读取 WANDB_PROJECT
  - 默认 project 仍然保持为 DiffSynth-Studio
  - 不改变 run name 或其他 logger 行为

  验证：

  - 已运行 python3 -m py_compile diffsynth/diffusion/
    logger.py

### CODE REVIEW

**Comment by Artiprocher** [comment]
@firefighter-eric Thanks for your contribution. We will add this parameter to the parsers immediately, and also support configure this parameter in SwanLab.

**Comment by firefighter-eric** [comment]
> @firefighter-eric Thanks for your contribution. We will add this parameter to the parsers immediately, and also support configure this parameter in SwanLab.

Thanks for merging this and for planning parser / SwanLab support.

One related follow-up request: it would be very helpful if DiffSynth could log more training runtime metrics to W&B / SwanLab, not only the basic training metrics.

For example:

- step time / iteration time
- data loading time if available
- forward / backward / optimizer step time if available
- checkpoint saving time
- checkpoint path / checkpoint size
- samples per second or images per second
- GPU memory allocated / reserved / peak memory
- global step, epoch, learning rate, loss, gradient norm

These metrics would make it much easier to monitor training performance and diagnose runtime bottlenecks.
